ajax/treebrowse.php

```diff
@@ -61,6 +61,7 @@
             }
         }
 
+        $params['treecriteria'] = count($params['criteria']);
```

Comment:
I had to read the code multiple time to understand what it the purpose of this.

Maybe you should add a key `'virtual' => true` to the criterai added below, and, before generating `$globallinkto`, removing all criteria having `'virtual' => true`.

This would make this logic more generic and may help us to add more "virtual" criteria.